ql/src/java/org/apache/hadoop/hive/ql/parse/QB.java

```diff
@@ -456,4 +456,22 @@ public boolean containsQueryWithoutSourceTable() {
     }
     return aliasToTabs.size()==0 && aliasToSubq.size()==0;
   }
+
+  // returns false when the query block doesn't have
+  // a table defined, e.g. "select 5"
+  public boolean hasTableDefined() {
+    // A table is not defined when there only exists a dummy
+    // table in the alias, so if it is blank, we know a table
+    // is defined.
+    if (aliasToTabs.size() == 0) {
+      return true;
+    }
+    for (String alias : aliasToTabs.keySet()) {
```

Comment:
Shouldn't we iterate through `tableAliases` as we do above the call to `hasTableDefined`?

Reply:
I looked at the code again...do you men aliases?  That would definitely be a better struct to use than the hashmap of tableAliases.  If you confirm this, I'll make the change.

Reply:
I was referring to `getTabAliases()`; I checked the code and realized that this is basically the same. This should be fine.